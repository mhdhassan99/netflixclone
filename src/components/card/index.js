import React, { useState, useContext, createContext } from 'react';

export const FeatureContext = createContext();

export default function Card({ children, ...restProps }) {
    const [showFeature, setShowFeature] = useState(false);
    const [itemFeature, setItemFeature] = useState({});

    return (
        <FeatureContext.Provider
            value={{ showFeature, setShowFeature, itemFeature, setItemFeature }}
        >
            <Container {...restProps}>{children}</Container>
        </FeatureContext.Provider>
    )
};

Card.Group = function CardGroup({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>
}