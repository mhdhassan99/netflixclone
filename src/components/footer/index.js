import React from 'react';

const Footer = ({ children, ...restProps }) => {
    return (
        <Container {...restProps}>
            {children}
        </Container>
    );
}
export default Footer;

Footer.Row = function FooterRow({ children, ...restProps }) {
    return <Row {...restProps}> {children} </Row>
} 

Footer.Column = function FooterColumn({ children, ...restProps }) {
    return <Column {...restProps}> {children} </Column>
} 

Footer.Link = function FooterLink({ children, ...restProps }) {
    return <Link {...restProps}> {children} </Link>
} 

Footer.Title = function FooterTitle({ children, ...restProps }) {
    return <Title {...restProps}> {children} </Title>
} 
