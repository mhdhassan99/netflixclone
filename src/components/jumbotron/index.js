import React from 'react';
import { Inner } from './styles/jumbotron';

const Jumbotron = ({children, direction = 'row', ...restProps }) => {
    return(
        <Inner direction={direction}>
            <p>I am Jumbo</p>
        </Inner>
    )
}

export default Jumbotron;
