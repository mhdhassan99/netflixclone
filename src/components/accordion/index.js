import React, {useState} from 'react';
import { Container, Inner } from '../jumbotron/styles/jumbotron';

const Accordion = ({ children, ...restProps }) => {
    return (
        <Container {...restProps}>
            <Inner>{children}</Inner>
        </Container>
    );
}
export default Accordion;

Accordion.Title = function AccordionTitle({ children, ...restProps}) {
    return <Title { ...restProps }>{children}</Title>
}; 

Accordion.Frame = function AccordionFrame({ children, ...restProps}) {
    return <Frame { ...restProps }>{children}</Frame>
}; 

Accordion.Item = function AccordionItem({ children, ...restProps}) {
    const [toggleShow, setToggleShow] = useState(false)

    return <Item { ...restProps }>{children}</Item>
}; 

Accordion.Header = function AccordionHeader({ children, ...restProps}) {
    return <Header { ...restProps }>{children}</Header>
}; 