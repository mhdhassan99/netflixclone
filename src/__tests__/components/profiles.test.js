import React from 'react';
import { render } from '@testing-library/react';
import { Profiles } from '../../components';

describe('<Profiles />', () => {
  it('renders the <Profiles /> with populated data', () => {
      
    const { container, getByText, getByTestId } = render (
      <Profiles>
        <Profiles.Title>Who's watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture 
                src="/images/john.png" 
                data-testid="profile-picture" 
            />
            <Profiles.Name>John</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );

    expect(getByText("Who's watching?"));
    expect(getByTestId('profile-picture')).toBeTruthy();
    expect(getByText("John"));
    expect(container.firstChild).toMatchSnapshot();
  });

});