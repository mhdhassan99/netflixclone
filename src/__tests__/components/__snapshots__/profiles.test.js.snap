// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> renders the <Profiles /> with populated data 1`] = `
<div
  class="profiles__Container-bxtkkv-0 fGoDNJ"
>
  <h1
    class="profiles__Title-bxtkkv-1 hPuemP"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 hybA-dO"
  >
    <li
      class="profiles__Item-bxtkkv-5 kJPhXM"
    >
      <img
        class="profiles__Picture-bxtkkv-4 kpLrAV"
        data-testid="profile-picture"
        src="/images/users//images/john.png.png"
      />
      <p
        class="profiles__Name-bxtkkv-3 jQTvVi"
      >
        John
      </p>
    </li>
  </ul>
</div>
`;
